fwprovider/access/resource_realm.go

```diff
@@ -0,0 +1,431 @@
+func NewRealmResource() resource.Resource {
+	return &realmResource{}
+}
+
+func (r *realmResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	cfg, ok := req.ProviderData.(config.Resource)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected config.Resource, got: %T", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = cfg.Client
+}
+
+func (r *realmResource) ConfigValidators(_ context.Context) []resource.ConfigValidator {
+	return []resource.ConfigValidator{
+		resourcevalidator.Conflicting(
+			path.MatchRoot("realm"),
+		),
+	}
+}
+
+func (r *realmResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages Realms on the Proxmox cluster",
+		MarkdownDescription: "Manages Realms on the Proxmox cluster.\n\n" +
+			"Realms are used for user authentication to a Proxmox node or cluster.\n",
+		Attributes: map[string]schema.Attribute{
+			"realm": schema.StringAttribute{
+				Description: "The ID of the realm which is of type string",
+				Required:    true,
+				Optional:    false,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"type": schema.StringAttribute{
+				Description: "Realm type",
+				Required:    true,
+				Optional:    false,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^ad$|^ldap$|^openid$|^pam$|^pve$`),
+						"must be one of [`ad`, `ldap`, `openid`, `pam`, `pve`]",
+					),
+				},
+			},
+			"acr_values": schema.StringAttribute{
+				Description: "Specifies the Authentication Context Class Reference values" +
+					"that theAuthorization Server is being requested to use for the Auth Request.",
+				Optional: true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^[^\x00-\x1F\x7F <>#"]*$`),
+						"Not a valid ACR value",
+					),
+				},
+			},
+			"autocreate": schema.BoolAttribute{
+				Description: "Automatically create users if they do not exist.\n" +
+					"default is 0 or false",
+				Optional: true,
+			},
+			"base_dn": schema.StringAttribute{
+				Description: "LDAP base domain name",
+				Optional:    true,
+			},
+			"bind_dn": schema.StringAttribute{
+				Description: "LDAP bind domain name",
+				Optional:    true,
+			},
+			"capath": schema.StringAttribute{
+				Description: "Path to the CA certificate store",
+				Optional:    true,
+			},
+			"case_sensitive": schema.BoolAttribute{
+				Description: "username is case-sensitive\n" +
+					"default is 1 or true",
+				Optional: true,
+			},
+			"cert": schema.StringAttribute{
+				Description: "Path to the client certificate",
+				Optional:    true,
+			},
+			"certkey": schema.StringAttribute{
+				Description: "Path to the client certificate key",
```

Comment:
should probably add `Sensitive: true` flag.
Not an expert, but I don't think this refers to a public key.